dattri/algorithm/data_shapley.py

```diff
@@ -0,0 +1,135 @@
+"""This module implement data shapley valuation."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+    from typing import List
+
+import torch
+from tqdm import tqdm
+
+from .base import BaseAttributor
+from .utils import _check_shuffle
+
+
+class KNNShalpeyAttributerExact(BaseAttributor):
+    """KNN Data Shapley Attributer."""
+
+    def __init__(
+        self,
+        k: int,
+        distance_func: Callable,
+    ) -> None:
+        """Initialize the AttributionTask.
+
+        KNN Data Shapley Valuation is task-agnostic.
```

Comment:
Task agnostic or model agnostic? I think the attribution/shapley values are still different for different test data points?

Reply:
In fact, one way to make KNN Shapley to be model dependent is to use the model hidden representation to calculate the distance between two data points.

Maybe we could add a task argument in __init__ so that we could extract some hidden representations of the model for distance calculation?

Reply:
I have currently updated the distance function with an Euclidean default. I would change the default to embedding distance after the `register_forward_hook` function is implemented.

Reply:
Sounds good. We should also change the description in the docstring here accordingly.